From aa397f0091d0e626f7ee3ec66db954c0b78c6806 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Subject: [PATCH] isofs: fix incorrect awk comparison operator

SEMICOLON = "YES" is not a correct comparison op (should use ==),
it's assignment! As a result, sub(";1$", "", name) was always executed.

Untested - I have no idea when SEMICOLON is not "YES"

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>.
---
 src/vfs/extfs/helpers/iso9660.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vfs/extfs/helpers/iso9660.in b/src/vfs/extfs/helpers/iso9660.in
index a7c891c..a06b732 100644
--- a/src/vfs/extfs/helpers/iso9660.in
+++ b/src/vfs/extfs/helpers/iso9660.in
@@ -153,7 +153,7 @@ BEGIN {
   sub("^----------   0    0    0", "-r--r--r--   1 root root", attr)
   sub(" $", "", name)
   # for Joliet UCS level 3
-  if (SEMICOLON = "YES") sub(";1$", "", name);
+  if (SEMICOLON == "YES") sub(";1$", "", name);
   ## sub(";[0-9]+$", "", name) ## would break copyout
   # skip . and ..
   if (name == ".") next;
-- 
1.8.1.4

